src/main/java/net/schmizz/sshj/sftp/RemoteFile.java

```diff
@@ -154,6 +154,9 @@ public void flush() throws IOException {
         @Override
         public void close() throws IOException {
             flush();
+            requester.request(newRequest(PacketType.CLOSE))
+                    .retrieve(requester.getTimeoutMs(), TimeUnit.MILLISECONDS)
+                    .ensureStatusPacketIsOK();
```

Comment:
I think you could just call `super.close()` instead of this.

Reply:
I don't think that would work due to this tid bit 
```
The close method of OutputStream does nothing
```

Reply:
found a way to clean it up @aldiyen. Good call!